lib/exp.py

```diff
@@ -67,13 +68,28 @@ def list_experiments(self):
                 EXPERIMENTS)
         return [Experiment.from_body(item) for item in response['items']]
 
-    def get_experiment(self, name):
-        response = self.k8s.get_namespaced_custom_object(
-                API,
-                API_VERSION,
-                self.namespace,
-                EXPERIMENTS,
-                name)
+    def get_experiment(self, name, max_retries=30):
+        retry_count = 0
+
+        # retry loop to wait for the experiment object to become available
+        while retry_count < max_retries:
+            try:
+                response = self.k8s.get_namespaced_custom_object(
```

Comment:
my biggest concern is that we're only adding retries for the `get_experiment()` call, but not any of the other calls that could run into the same kubernetes object availability timing issue (list, create, update, delete).  

If we've agreed that introducing retries is the right approach here, my suggestion would be to instead implement retry functionality in one place and have each of the `k8s.
*_namespaced_custom_object()` calls optionally leverage this.